### PR TITLE
Inheritance fixes

### PR DESCRIPTION
**Pull request by jogibear9988**

### CODE REVIEW

**Comment by jogibear9988** [comment]
@sdanyliv @MaceWindu also done, if anyone can review?

**Comment by jogibear9988** [comment]
this one? ;-)

File: Tests/Model/ParentChild.cs [hunk 1]
```diff
@@ -507,7 +507,6 @@ public class ParentInheritance12 : ParentInheritanceBase2
 	#region Inheritance3
 
 	[Table(Name="Parent")]
-	[InheritanceMapping(Code = null, Type = typeof(ParentInheritanceBase3))]
```
Comment (sdanyliv) [inline Tests/Model/ParentChild.cs]: Why you have removed this mapping from test?

**Comment by jogibear9988** [comment]
cause it‘s invalid!
base class is abstract, and so no instance of it can be created!
we have to remove the abstract or the mapping!

File: Source/LinqToDB/Expressions/ConvertFromDataReaderExpression.cs [hunk 2]
```diff
@@ -151,7 +151,22 @@ public object GetValue(IDataReader dataReader)
...
 				}
 
-				return func(dataReader);
+				try
+				{
+					return func(dataReader);
+				}
+				catch (FormatException ex)
+				{
+					throw new FormatException("FormatException on Column: " + dataReader.GetName(_columnIndex), ex);
```
Comment (MaceWindu) [inline Source/LinqToDB/Expressions/ConvertFromDataReaderExpression.cs]: Why do we need 3 cases here? I think one rethrow is enough

Comment (jogibear9988) [reply Source/LinqToDB/Expressions/ConvertFromDataReaderExpression.cs]: I wanted to keep the original exception type!
What type of exception should I throw?

Comment (MaceWindu) [reply Source/LinqToDB/Expressions/ConvertFromDataReaderExpression.cs]: Well, I don't think it makes sense to mimic original exception. We just can throw LinqToDbException like "mapping of column {name} value failed, see inner exception for details"

**Pull request merged**